lib/types/utils.ts

```diff
@@ -1,14 +1,19 @@
-type SysWithId = {
+type Sys = {
   id: string
+  type: string
+  [key: string]: any
 }
 
-export type WithOptionalId<Type extends { sys: unknown }> = Omit<Type, 'sys'> & {
-  sys?: Partial<Omit<Type['sys'], 'id'> & SysWithId>
+type Entity = {
+  sys: Sys
+  [key: string]: any
 }
 
-export type WithId<Type extends { sys: unknown }> = Omit<Type, 'sys'> & {
-  sys: Partial<Omit<Type['sys'], 'id'>> & SysWithId
-}
+type Optional<Type, Keys extends keyof Type> = Partial<Type> & Omit<Type, Keys>
+
+export type WithSysWithAtLeastId = { sys: { id: string; [key: string]: any } }
+export type WithId<Type extends Entity> = Omit<Type, 'sys'> & WithSysWithAtLeastId
```

Comment:
This drops all properties of `Type.sys` except of `id`. I'd therefore do something like this and remove line 14.

```suggestion
export type WithId<Type extends Entity> = Omit<Type, 'sys'> & { sys: Omit<Entity['sys'], 'id'> & Required<Pick<Entity['sys'], 'id'>> }
```

That way the properties don't get lost.

Reply:
Well, in this case we purposely want the properties to get lost because we ignore them in the underlying calls (this also answers the the comment above about using string-indexed access)